### PR TITLE
test: Add _getEarlyPaymentBreakdown() test (SC-4381)

### PR DESCRIPTION
**Pull request by edag94**
# Description
Add test for _getEarlyPaymentBreakdown() in MapleLoanInternals

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/test/MapleLoanInternals.t.sol [hunk 1]
```diff
@@ -1750,7 +1750,46 @@ contract MapleLoanInternals_AcceptNewTermsTests is TestUtils {
...
+        _token1 = new MockERC20("Token1", "T1", 0);
+
+        _defaultBorrower    = address(1);
+        _defaultAssets      = [address(_token0), address(_token1)]; 
+        _defaultTermDetails = [uint256(1), uint256(2), uint256(3)];
+        _defaultAmounts     = [uint256(5), uint256(100), uint256(0)];
+        _defaultRates       = [uint256(.05 ether), uint256(.1 ether), uint256(.15 ether), uint256(20)];
+
+        _loan.initialize(_defaultBorrower, _defaultAssets, _defaultTermDetails, _defaultAmounts, _defaultRates);
+        _loan.setPrincipal(_defaultAmounts[1]);
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1780]: We should move these three into the test

File: contracts/test/MapleLoanInternals.t.sol [hunk 2]
```diff
@@ -1750,7 +1750,46 @@ contract MapleLoanInternals_AcceptNewTermsTests is TestUtils {
...
+    }
+
+    function test_getEarlyPaymentBreakdown() external {
+        (uint256 principal, uint256 interest) = _loan.getEarlyPaymentBreakdown();
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1784]: `( uint256 principal, uint256 interest ) = _loan.getEarlyPaymentBreakdown(); `

File: contracts/test/MapleLoanInternals.t.sol [hunk 3]
```diff
@@ -1750,7 +1750,46 @@ contract MapleLoanInternals_AcceptNewTermsTests is TestUtils {
...
+
+        _loan.initialize(_defaultBorrower, _defaultAssets, _defaultTermDetails, _defaultAmounts, _defaultRates);
+        _loan.setPrincipal(_defaultAmounts[1]);
+    }
+
+    function test_getEarlyPaymentBreakdown() external {
+        (uint256 principal, uint256 interest) = _loan.getEarlyPaymentBreakdown(); 
+
+        uint256 expectedPrincipal = _defaultAmounts[1];
+        uint256 expectedInterest = expectedPrincipal * _defaultRates[1] / 10**18;
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1787]: Align equals

Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1787]: `10 ** 18`

File: contracts/test/MapleLoanInternals.t.sol [hunk 4]
```diff
@@ -1750,7 +1750,46 @@ contract MapleLoanInternals_AcceptNewTermsTests is TestUtils {
...
+
+        assertEq(principal, expectedPrincipal);
+        assertEq(interest, expectedInterest);
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1790]: Align params

File: contracts/test/MapleLoanInternals.t.sol [hunk 5]
```diff
@@ -1750,7 +1750,46 @@ contract MapleLoanInternals_AcceptNewTermsTests is TestUtils {
...
+        _defaultAssets      = [address(_token0), address(_token1)]; 
+        _defaultTermDetails = [uint256(1), uint256(2), uint256(3)];
+        _defaultAmounts     = [uint256(5), uint256(100), uint256(0)];
+        _defaultRates       = [uint256(.05 ether), uint256(.1 ether), uint256(.15 ether), uint256(20)];
+
+        _loan.initialize(_defaultBorrower, _defaultAssets, _defaultTermDetails, _defaultAmounts, _defaultRates);
+        _loan.setPrincipal(_defaultAmounts[1]);
+    }
+
+    function test_getEarlyPaymentBreakdown() external {
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1783]: Should fuzz principalRequested and earlyFee params

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: contracts/test/MapleLoanInternals.t.sol [hunk 6]
```diff
@@ -1768,7 +1768,59 @@ contract MapleLoanInternals_AcceptNewTermsTests is TestUtils {
...
+        
+        // Set _initialize() parameters.
+        _token0 = new MockERC20("Token0", "T0", 0);
+        _token1 = new MockERC20("Token1", "T1", 0);
+
+        _defaultBorrower    = address(1);
+        _defaultAssets      = [address(_token0), address(_token1)]; 
+        _defaultTermDetails = [uint256(1), uint256(2), uint256(3)];
+        _defaultAmounts     = [uint256(5), uint256(100), uint256(0)];
+        _defaultRates       = [uint256(.05 ether), uint256(.1 ether), uint256(.15 ether), uint256(20)];
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1797]: Lets bring these into the test itself instead of overriding the values

Comment (lucas-manuel) [reply contracts/test/MapleLoanInternals.t.sol:1797]: Also should use `uint256(0.05 ether)` etc

File: contracts/test/MapleLoanInternals.t.sol [hunk 7]
```diff
@@ -1768,7 +1768,59 @@ contract MapleLoanInternals_AcceptNewTermsTests is TestUtils {
...
+        _defaultTermDetails = [uint256(1), uint256(2), uint256(3)];
+        _defaultAmounts     = [uint256(5), uint256(100), uint256(0)];
+        _defaultRates       = [uint256(.05 ether), uint256(.1 ether), uint256(.15 ether), uint256(20)]; 
+    }
+
+    function test_getEarlyPaymentBreakdown(uint256 principal_, uint256 earlyFeeRate_) external {
+        uint256 maxEarlyFeeRateForTestCase = 1 * SCALED_ONE; // 100%
+
+        principal_    = constrictToRange(principal_, 1, type(uint256).max / maxEarlyFeeRateForTestCase);
+        earlyFeeRate_ = constrictToRange(earlyFeeRate_, 1, maxEarlyFeeRateForTestCase);
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1804]:
```js
        principal_    = constrictToRange(principal_,    1, type(uint256).max / maxEarlyFeeRateForTestCase);
        earlyFeeRate_ = constrictToRange(earlyFeeRate_, 1, maxEarlyFeeRateForTestCase);

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: contracts/test/MapleLoanInternals.t.sol (hunk 6 above)
Comment (edag94) [reply contracts/test/MapleLoanInternals.t.sol:1797]: Ok i was just staying consistent with the way i had the setup for another test suite. I can change it to set in test specifically

**Pull request merged**